### PR TITLE
Save Result Data to PDF or CSV

### PR DESCRIPTION
**Pull request by pixelsaurus**
Closes: #563

### CODE REVIEW

File: packages/coinstac-ui/app/render/components/results/displays/images.jsx [hunk 1]
```diff
@@ -37,37 +69,176 @@ class Images extends Component {
...
+          item.push(
+            <img alt={`${v}-img`} src={`data:image/png;base64, ${v}`}
+            style={styles.image} />
+          );
         } else {
-          Object.entries(v).forEach(([l, w]) => {
-            output.push(<h4 key={`image-${k}-${l}`}>{this.humanize(l).replace('.png', '')}</h4>);
+          Object.entries(v).forEach(([l, w], h) => {
+            let subItemsLength = Object.entries(v).length;
+            console.log(subItemsLength);
```
Comment (rssk) [inline packages/coinstac-ui/app/render/components/results/displays/images.jsx]: `con solo` 🔫

File: packages/coinstac-ui/app/render/components/results/displays/images.jsx [hunk 2]
```diff
@@ -37,37 +69,176 @@ class Images extends Component {
...
+            let canvas = key + '_canvas';
+            let page = 'page-' + key;
+            let key_canvas = ReactDOM.findDOMNode(this.refs[canvas]);
+            let canvasImg = key_canvas.toDataURL("image/jpg", 1.0);
+            doc.addPage();
+      		  doc.addImage(canvasImg, 'jpg', 5, 5, 200, height);
+        });
+  		  doc.save(this.slugify(this.props.title) + ".pdf");
+  		 } catch(e) {
+  			 alert("Error description: " + e.message);
```
Comment (rssk) [inline packages/coinstac-ui/app/render/components/results/displays/images.jsx]: does this use the error notifications? Or does this just popup a literal alert?

**Review by rssk — CHANGES_REQUESTED** [review]
Looks pretty great, just two issues

File: packages/coinstac-ui/app/render/components/results/displays/images.jsx (hunk 1 above)
Comment (swashcap) [reply packages/coinstac-ui/app/render/components/results/displays/images.jsx]: Gotta add “no console” to the linter!

File: packages/coinstac-ui/app/render/components/results/displays/images.jsx [hunk 3]
```diff
@@ -27,6 +43,24 @@ class Images extends Component {
...
+    /**
+     * slugify
+     * Converts human readable string to a slug
+     * @param  {string} text Input String to Convert
+     * @return {string}     Converted String to Slug
+     */
+
+     // ignore weird class-methods-use-this lint error
+     // eslint-disable-next-line
+    slugify(text) {
```
Comment (swashcap) [inline packages/coinstac-ui/app/render/components/results/displays/images.jsx]: [Lodash's `kebabCase`](https://lodash.com/docs/4.17.10#kebabCase) does exactly this.

File: packages/coinstac-ui/app/render/components/results/displays/images.jsx [hunk 4]
```diff
@@ -37,48 +71,199 @@ class Images extends Component {
...
+        let key_results = ReactDOM.findDOMNode(this.refs[page]);
+        RasterizeHTML.drawHTML(key_results.innerHTML, key_canvas);
+    });
+  }
+
+  componentDidMount = () => {
+    setTimeout(this.renderCanvas, 1000);
+  }
+
+  savePDF = () => {
```
Comment (swashcap) [inline packages/coinstac-ui/app/render/components/results/displays/images.jsx:176]: Wow, surprisingly little code to get a PDF! That's neat.

File: packages/coinstac-ui/app/render/components/results/displays/result-table.jsx [hunk 5]
```diff
@@ -1,28 +1,73 @@
...
+
+/**
+ * exportTableToCSV
+ * Convert HTML Table to CSV formatted string
+ * @param {string} html Table hypertext
+ * @param {string} filename The name of the file
+ */
+function exportTableToCSV(html, filename) {
+  const csv = [];
+  const rows = document.querySelectorAll('table tr');
```
Comment (swashcap) [inline packages/coinstac-ui/app/render/components/results/displays/result-table.jsx:34]: Do you need DOM APIs to retrieve the data? If your component already has all access to the renderable data it should be able to map it into a CSV. Just a thought.

**Review by swashcap — COMMENTED** [review]
:wave: Hello! I used to work on this project and noticed this PR in my GitHub notifications. Sounds like a neat feature! I had a couple of comments that might help you out.

File: packages/coinstac-ui/app/render/components/results/displays/images.jsx (hunk 3 above)
Comment (pixelsaurus) [reply packages/coinstac-ui/app/render/components/results/displays/images.jsx]: Thanks for the tip. Changed to Lodash.

File: packages/coinstac-ui/app/render/components/results/displays/images.jsx (hunk 4 above)
Comment (pixelsaurus) [reply packages/coinstac-ui/app/render/components/results/displays/images.jsx:176]: Thanks!

File: packages/coinstac-ui/app/render/components/results/displays/result-table.jsx (hunk 5 above)
Comment (pixelsaurus) [reply packages/coinstac-ui/app/render/components/results/displays/result-table.jsx:34]: This is true, however the goal was to preserve the unique table formatting for the end user, which would have to be fully duplicated if I pulled from the results object.

**Comment by pixelsaurus** [comment]
Fixed merge conflicts

**Review by rssk — APPROVED** [review]

**Pull request merged**